test/unittests/execute_control_test.cpp

```diff
@@ -613,3 +613,157 @@ TEST(execute_control, return_from_block_stack_cleanup)
     ASSERT_EQ(ret.size(), 1);
     EXPECT_EQ(ret[0], 1);
 }
+
+TEST(execute_control, if_smoke)
+{
+    /*
+    (func (param $x i32) (result i32)
+      (local $y i32)
+      get_local $x
+      (if
+        (then
+          i32.const 4
+          set_local $y
+        )
+      )
+      get_local $y
+    )
+    */
+    const auto bin = from_hex(
+        "0061736D0100000001060160017F017F030201000A11010F01017F20000440410421010B20010B0010046E616D"
+        "650209010002000178010179");
+
+    auto module = parse(bin);
+
+    for (const auto param : {0u, 1u})
+    {
+        constexpr uint64_t expected_results[]{
+            0,  // no if branch.
+            4,  // if branch.
+        };
+
+        const auto [trap, ret] = execute(module, 0, {param});
+        ASSERT_FALSE(trap);
+        ASSERT_EQ(ret.size(), 1);
+        EXPECT_EQ(ret[0], expected_results[param]);
+    }
+}
+
+TEST(execute_control, if_else_smoke)
+{
+    /*
+    (func (param $x i32) (result i32)
+      get_local $x
+      (if (result i32)
+        (then
+          i32.const 1
+        )
+        (else
+          i32.const 2
+        )
+      )
+    )
+    */
+    const auto bin = from_hex(
+        "0061736D0100000001060160017F017F030201000A0E010C002000047F41010541020B0B000D046E616D650206"
+        "010001000178");
+
+    auto module = parse(bin);
```

Comment:
```suggestion
    const auto module = parse(bin);
```
I believe `module` can be `const`. If I'm right, update also other cases.

Reply:
Done.